ITP should hold off longer on deleting website data for domains with WebPush that the user interacts with

johnwilander opened the pull request:
#### c5e9bfec8ca40fa0224df9b8a47638953864edcd
<pre>
ITP should hold off longer on deleting website data for domains with WebPush that the user interacts with
<a href="https://bugs.webkit.org/show_bug.cgi?id=244609">https://bugs.webkit.org/show_bug.cgi?id=244609</a>
&lt;rdar://99148432&gt;

Reviewed by Alex Christensen.

* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsDatabaseStore.cpp:
(WebKit::ResourceLoadStatisticsDatabaseStore::hasHadRecentWebPushInteraction const):
    New convenience function.
(WebKit::ResourceLoadStatisticsDatabaseStore::registrableDomainsToDeleteOrRestrictWebsiteDataFor):
    Now checks hasHadRecentWebPushInteraction() and exempts the domain if true.
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsDatabaseStore.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
    Updated the ITPCleanup test to cover the new behavior.

Canonical link: <a href="https://commits.webkit.org/254048@main">https://commits.webkit.org/254048@main</a>
</pre>

achristensen07 reviewed (approved):
Looks good.  I verified the two tests failing in the old EWS are fixed by the new push.  I don't know how to kick EWS to go again, though.

johnwilander committed:
ITP should hold off longer on deleting website data for domains with WebPush that the user interacts with
https://bugs.webkit.org/show_bug.cgi?id=244609
<rdar://99148432>

Reviewed by Alex Christensen.

* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsDatabaseStore.cpp:
(WebKit::ResourceLoadStatisticsDatabaseStore::hasHadRecentWebPushInteraction const):
    New convenience function.
(WebKit::ResourceLoadStatisticsDatabaseStore::registrableDomainsToDeleteOrRestrictWebsiteDataFor):
    Now checks hasHadRecentWebPushInteraction() and exempts the domain if true.
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsDatabaseStore.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
    Updated the ITPCleanup test to cover the new behavior.

Canonical link: https://commits.webkit.org/254048@main
